src/modules/launcher/Plugins/Microsoft.Plugin.Calculator/CalculateHelper.cs

```diff
@@ -41,6 +41,13 @@ public static bool InputValid(string input)
                 return false;
             }
 
+            // If the input ends with a binary operator then it is not a valid input to mages and the Interpret function would throw an exception.
+            string trimmedInput = input.TrimEnd();
+            if (trimmedInput.EndsWith('+') || trimmedInput.EndsWith('-') || trimmedInput.EndsWith('*') || trimmedInput.EndsWith('|') || trimmedInput.EndsWith('\\') || trimmedInput.EndsWith('^') || trimmedInput.EndsWith('=') || trimmedInput.EndsWith('&'))
```

Comment:
Can we make this part of regex itself?

Reply:
Unfortunately a complete mathematical expression cannot be expressed perfectly using a regex. we need a BNF parser which does this and it cannot be unambiguously written in C#. Ref - https://stackoverflow.com/questions/153572/parser-generator-that-outputs-c-sharp-given-a-bnf-grammar
The reason why it's not possible to have it as a part of the regex is because it's recursive.
We want something along these lines - 

```
assignable_expr ::= identifier | assignment | member | call
computing_expr ::= literal | binary | unary | function | literal | condition | range | arguments
expr ::= assignable_expr | computing_expr
assignment ::= assignable_expr space* '=' space* expr
```
so assignment uses assignable_expr which again uses assignment and so on, which is not possible using a regex. Hence, I had to add a check that the input does not end with a binary operator.